src/instructions/common.rs

```diff
@@ -80,13 +81,26 @@ pub fn addiw<Mac: Machine>(
 // =======================
 // #  LOAD instructions  #
 // =======================
+fn check_load_boundary<R: Register>(version0: bool, address: &R, bytes: u64) -> Result<(), Error> {
+    if version0 {
+        let address = address.to_u64();
+        let end = address.checked_add(bytes).ok_or(Error::OutOfBound)?;
+        if end == RISCV_MAX_MEMORY as u64 {
```

Comment:
I am curious why it is not `end >= RISCV_MAX_MEMORY`

Reply:
The `>` case will be covered by normal memory module bound checking code. This bug only occurs with the `==` case.